accountClient/accountClient_test.go

```diff
@@ -158,3 +160,58 @@ func TestInternalGetClientAccountReturnsClientsAccountId(t *testing.T) {
 		t.Errorf("Expected account id to be %s, got %s", accountID, account.AccountID)
 	}
 }
+
+func TestInitEmailUpdateWithValidAccountSucceeds(t *testing.T) {
+	// make internal account client (v1)
+	registrationClient := accountClient.New(ValidClientConfig)
+	_, resp, err := test.MakeE3DBAccount(t, &registrationClient, uuid.New().String(), e3dbAuthHost)
+	if err != nil {
+		t.Fatalf("Could not register account %s\n", err)
+	}
+	currentTime := time.Now().Local()
+	// make internal account client (v2)
+	updateClient := accountClient.NewV2(ValidClientConfigV2)
+	newEmailReq := accountClient.InitiateUpdateEmailRequest{
+		AccountID:    resp.Profile.AccountID,
+		CurrentEmail: resp.Profile.Email,
+		NewEmail:     "test" + uuid.New().String() + "@example.com",
+		CreatedAt:    currentTime,
+		CoolOffEnd:   currentTime.Add(time.Hour * 24),
+	}
+	// Make request to post / initiate the  email update
+	response, err := updateClient.InitiateEmailUpdate(testCtx, newEmailReq)
+	if err != nil {
+		t.Fatalf("Failed to initiate email update \n Email Req: (%+v) \n error %+v", newEmailReq, err)
+	}
+	// Verify that the AccountID returned matches the client
+	if response.AccountID.String() != newEmailReq.AccountID {
+		t.Fatalf("AccountID (%+v) passed in does not match AccountID in response (%+v) for request (%+v)", response.AccountID.String(), newEmailReq.AccountID, newEmailReq)
+	}
+}
+
+func TestInitEmailUpdateWithTwoReqsFails(t *testing.T) {
+	registrationClient := accountClient.New(ValidClientConfig)
+	_, resp, err := test.MakeE3DBAccount(t, &registrationClient, uuid.New().String(), e3dbAuthHost)
+	if err != nil {
+		t.Fatalf("Could not register account %s\n", err)
+	}
+	currentTime := time.Now().Local()
+	updateClient := accountClient.NewV2(ValidClientConfigV2)
+	newEmailReq := accountClient.InitiateUpdateEmailRequest{
+		AccountID:    resp.Profile.AccountID,
+		CurrentEmail: resp.Profile.Email,
+		NewEmail:     "test" + uuid.New().String() + "@example.com",
+		CreatedAt:    currentTime,
+		CoolOffEnd:   currentTime.Add(time.Hour * 24),
+	}
+	// Make request to post / initiate the email update & verify it's successful
+	_, err = updateClient.InitiateEmailUpdate(testCtx, newEmailReq)
+	if err != nil {
+		t.Fatalf("Failed to initiate email update \n Email Req: (%+v) \n error %+v", newEmailReq, err)
+	}
+	// Make request to post / initiate the same email update again & verify it fails
+	_, err = updateClient.InitiateEmailUpdate(testCtx, newEmailReq)
+	if err == nil {
```

Comment:
(please talk to @rosasck / look at some of the tests for storageClient) on how to verify that not only an error was returned but that that status code is the expected value (assuming [409](https://http.cat/409) for this type of error)